ocean_lib/exceptions.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+class ContractNotFound(Exception):
+    """
+    Contract address is not found
+    in the factory events
+    """
+
+
```

Comment:
It's best that errors aren't named based on a **location** (module) for the error since that information is already provided in the stack trace; but rather the error is more the **type** of issue. 

DDOError, BPoolError, and ExchangeError only give a location, whereas e.g. ValidationError is the type.

Therefore many of my suggestions below are about renaming to specific types of errors. By the time we're done, there shouldn't be any location-based error classes (DDOError, BPoolError, and ExchangeError).